ebpf/c/tracer-ebpf.c

```diff
@@ -20,83 +20,83 @@
-	.type = BPF_MAP_TYPE_HASH,
-	.key_size = sizeof(ipv4_tuple_t),
-	.value_size = sizeof(conn_stats_ts_t),
-	.max_entries = 65536,
-	.pinning = 0,
-	.namespace = "",
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(ipv4_tuple_t),
+    .value_size = sizeof(conn_stats_ts_t),
+    .max_entries = 65536,
+    .pinning = 0,
+    .namespace = "",
 };
 
 /* This is a key/value store with the keys being an ipv6_tuple_t for send & recv calls
  * and the values being the struct conn_stats_ts_t *.
  */
 struct bpf_map_def SEC("maps/conn_stats_ipv6") conn_stats_ipv6 = {
-	.type = BPF_MAP_TYPE_HASH,
-	.key_size = sizeof(ipv6_tuple_t),
-	.value_size = sizeof(conn_stats_ts_t),
-	.max_entries = 65536,
-	.pinning = 0,
-	.namespace = "",
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(ipv6_tuple_t),
+    .value_size = sizeof(conn_stats_ts_t),
+    .max_entries = 65536,
+    .pinning = 0,
+    .namespace = "",
 };
 
 /* These maps are used to match the kprobe & kretprobe of connect for IPv4 */
 /* This is a key/value store with the keys being a pid
  * and the values being a struct sock *.
  */
 struct bpf_map_def SEC("maps/connectsock_ipv4") connectsock_ipv4 = {
-	.type = BPF_MAP_TYPE_HASH,
-	.key_size = sizeof(__u64),
-	.value_size = sizeof(void *),
-	.max_entries = 1024,
-	.pinning = 0,
-	.namespace = "",
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(__u64),
+    .value_size = sizeof(void*),
+    .max_entries = 1024,
+    .pinning = 0,
+    .namespace = "",
 };
 
 /* These maps are used to match the kprobe & kretprobe of connect for IPv6 */
 /* This is a key/value store with the keys being a pid
  * and the values being a struct sock *.
  */
 struct bpf_map_def SEC("maps/connectsock_ipv6") connectsock_ipv6 = {
-	.type = BPF_MAP_TYPE_HASH,
-	.key_size = sizeof(__u64),
-	.value_size = sizeof(void *),
-	.max_entries = 1024,
-	.pinning = 0,
-	.namespace = "",
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(__u64),
+    .value_size = sizeof(void*),
+    .max_entries = 1024,
+    .pinning = 0,
+    .namespace = "",
 };
 
 /* This map is used to match the kprobe & kretprobe of udp_recvmsg */
 /* This is a key/value store with the keys being a pid
  * and the values being a struct sock *.
  */
 struct bpf_map_def SEC("maps/udp_recv_sock") udp_recv_sock = {
-	.type = BPF_MAP_TYPE_HASH,
-	.key_size = sizeof(__u64),
-	.value_size = sizeof(void *),
-	.max_entries = 1024,
-	.pinning = 0,
-	.namespace = "",
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(__u64),
+    .value_size = sizeof(void*),
+    .max_entries = 1024,
+    .pinning = 0,
+    .namespace = "",
 };
 
 /* http://stackoverflow.com/questions/1001307/detecting-endianness-programmatically-in-a-c-program */
-__attribute__((always_inline))
-static bool is_big_endian(void) {
-	union {
-		uint32_t i;
-		char c[4];
-	} bint = {0x01020304};
-
-	return bint.c[0] == 1;
+__attribute__((always_inline)) static bool is_big_endian(void)
```

Comment:
This was my blocker when I tried using `clang-format` a while back - are you able to find any configuration that puts the `__attribute__` on its own line?

Reply:
I will look at that 👍